lib/fs.js

```diff
@@ -2023,22 +2023,25 @@ function ReadStream(path, options) {
     }
 
     this.pos = this.start;
-  }
-
-  // Backwards compatibility: Make sure `end` is a number regardless of `start`.
-  // TODO(addaleax): Make the above typecheck not depend on `start` instead.
-  // (That is a semver-major change).
-  if (typeof this.end !== 'number')
-    this.end = Infinity;
+  } else {
+    // Backwards compatibility: Make sure `end` is a number regardless of
+    // `start`.
+    // TODO(addaleax): Make the above typecheck not depend on `start` instead.
+    // (That is a semver-major change).
+    if (typeof this.end !== 'number')
+      this.end = Infinity;
+    else if (!Number.isInteger(this.end))
```

Comment:
The reason that the `else` block above became necessary is the usage of `Number.isInteger` as it returns false for `Infinity`. That was a valid input before and will now result in an error. I personally am OK with not accepting that as valid input but others might disagree. Ping @addaleax
This should only be important for `this.end` as settings `this.start` does not make any sense to set to `Infinity`.

Reply:
Don't know if the API documentation was accepting passing `Infinity` for `end`, but it doesn't now, as the doc states they are both integers (https://nodejs.org/dist/latest-v9.x/docs/api/fs.html). Need @addaleax to check if it was.